### DISCUSSION TITLE
How can i set username/password in TeslaMate-Mosquitto-MQTT-Broker?

### DISCUSSION THREAD

**Discussion by abc123xxx** [teslamate-org/teslamate#2350] [Q&A]
How can i set username/password in the default TeslaMate-Mosquitto-MQTT-Broker?

With environment-variables
MQTT_USERNAME
MQTT_PASSWORD
it not worked.
I can always connect with a free MQTT-Client without username/password!

**Comment by abc123xxx** [comment]
Does anyone have an idea or instructions on how to protect the Mosquitto broker with a username/password?

**Comment by adriankumpf** [comment]
Check out the documentation over at their Docker Hub page: https://hub.docker.com/_/eclipse-mosquitto

The _mosquitto.conf_ is documented here https://mosquitto.org/man/mosquitto-conf-5.html

**Comment by abc123xxx** [comment]
And why it is not working with the TeslaMate environment-variables:
MQTT_USERNAME
MQTT_PASSWORD

**Reply by DjAnu** [reply]
Hi I am also looking to find a secure method to expose MQTT server to internet, Were you able to configure your mqtt server with user name / passed ?

**Reply by brianmay** [reply]
Those environment variables work fine for me. No idea why it isn't working for you.

(maybe check you can connect with those credentials with another client, sometimes the mosquitto server can be a little bit non-obvious how it works)

**Reply by DjAnu** [reply]
Lol, I was actually asking how do I set up login credentials for MQTT server. Currrenly I don't use any login / passed. I followed this guide to set up Teslamate but now I want to use same MQTT server for some other applications as well from outside my local network.

**Reply by lukasradek** [reply]
same here :)

**Reply by lukasradek** [reply]
@DjAnu  have you specified the variables only in the .env file or have you also put them into the docker compose? That is the means of passing them to the container.
